### PR TITLE
oc: add --insecure-policy for creating edge routes

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1354377

@fabianofranz PTAL

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

File: pkg/cmd/cli/cmd/create/route.go [hunk 1]
```diff
@@ -130,6 +131,11 @@ func CreateEdgeRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, ar
 	}
 	route.Spec.TLS.CACertificate = string(caCert)
 
+	insecurePolicy := kcmdutil.GetFlagString(cmd, "insecure-policy")
+	if len(insecurePolicy) > 0 {
+		route.Spec.TLS.InsecureEdgeTerminationPolicy = api.InsecureEdgeTerminationPolicyType(insecurePolicy)
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/create/route.go]: Does it validate correctly against invalid values in `insecurePolicy`?

**Comment by fabianofranz** [comment]
Minor comment otherwise LGTM.

File: pkg/cmd/cli/cmd/create/route.go (hunk 1 above)
Comment (kargakis) [inline pkg/cmd/cli/cmd/create/route.go]: Yes.
```
$ oc create route edge --service=foo --insecure-policy=Allowed --port=80
The Route "foo" is invalid.
spec.tls.insecureEdgeTerminationPolicy: Invalid value: "Allowed": invalid value for InsecureEdgeTerminationPolicy option, acceptable values are None, Allow, Redirect, or empty
```

**Comment by kargakis** [comment]
Comment addressed [merge]

**Comment by kargakis** [comment]
```
FAILURE after 119.223s: test/end-to-end/core.sh:29: executing 'oc get -n test pods' expecting any result and text 'frontend.+Running'; re-trying every 0.2s until completion or 120.000s: the command timed out
```
It's taking more than 2m for the deployer to be created
```
frontend-1-deploy           0/1       ContainerCreating   0          2m
```
@smarterclayton @mfojtik

**Comment by kargakis** [comment]
[merge]

**Comment by smarterclayton** [comment]
The frontend problem is the same as the database - secret binding is the
problem.

**Pull request merged**